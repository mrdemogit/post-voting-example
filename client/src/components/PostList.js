import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Post from './Post'
import { PostType } from '../types'

const StyledList = styled.div`
  padding: 1em;
  font-family: 'Roboto';
  font-size: 0.9em;
`

const PostList = ({ posts, upvote, downvote, error }) => {
  return (
    <StyledList>
      {error}
      {posts.map((post) => (
        <Post key={post.uuid} {...post} upvote={upvote} downvote={downvote} />
      ))}
    </StyledList>
  )
}

PostList.propTypes = {
  posts: PropTypes.arrayOf(PropTypes.shape(PostType)),
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired,
  error: PropTypes.string
}

export default PostList
