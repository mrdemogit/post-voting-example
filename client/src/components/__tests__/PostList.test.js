import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import PostList from '../PostList'

describe('PostList', () => {
  it('should render correctly', () => {
    const renderer = new ShallowRenderer()
    renderer.render(
      <PostList
        posts={[
          {
            uuid: 'uuid1',
            author: 'author1',
            content: 'contentTest1',
            votes: -1,
            comments: [
              {
                uuid: 'uuid13',
                author: 'author13',
                content: 'contentTest13',
                votes: 13
              }
            ]
          },
          {
            uuid: 'uuid3',
            author: 'author3',
            content: 'contentTest3',
            votes: 0,
            comments: []
          }
        ]}
        upvote={() => {}}
        downvote={() => {}}
      />
    )

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
