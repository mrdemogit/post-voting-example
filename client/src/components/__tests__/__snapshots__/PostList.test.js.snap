// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList should render correctly 1`] = `
<ForwardRef>
  <UNDEFINED
    author="author1"
    comments={
      Array [
        Object {
          "author": "author13",
          "content": "contentTest13",
          "uuid": "uuid13",
          "votes": 13,
        },
      ]
    }
    content="contentTest1"
    downvote={[Function]}
    upvote={[Function]}
    uuid="uuid1"
    votes={-1}
  />
  <UNDEFINED
    author="author3"
    comments={Array []}
    content="contentTest3"
    downvote={[Function]}
    upvote={[Function]}
    uuid="uuid3"
    votes={0}
  />
</ForwardRef>
`;
